_episodes/01-introduction.md

```diff
@@ -4,14 +4,105 @@ teaching: 15
 objectives:
 - "identify basic concepts in programming"
 keypoints:
 - 
 ---
-FIXME
+
+## Programming in Python
+
+In most general terms, programming is the process of writing instructions for a computer. In this course we will be using Python as the language to communicate with the computer. 
+
+> Strictly speaking, Python is an interpreted language, rather than a compiled language, meaning we are not communicating directly with the computer when we use Python.
+
+Programming is a wide topic including a variety of techniques and tools. In this course we'll be focusing on programming for statistical analysis.
+
+
+### IDEs
+
+IDE stands for Integrated Development Environment. IDEs are where you will write, edit, and debug python scripts, so you want to choose one that makes you feel comfortable and includes the functionality that you need. 
+
+### Packages
+
+Packages, or libraries, are extensions to the statistical programming language. They contain code, data, and documentation in a standardised collection format that can be installed by users, typically via a centralised software repository.
+
+## Best Practices
+
+Let's overview some base concepts that any programmer should always keep in mind.
+
+### Documentation
+
+Have you ever returned to a task and tried to read a note that you quickly scrawled for yourself the last time you were working on it? Have you ever inherited a project from a colleague and foudn you have no idea what remains to be done?
+
+It can be very challenging to return to your own work or a colleague's and this goes doubly for programming. Documentation is one way we can reduce the burden on future selves and our colleagues.
+
+#### Inline Documentation
+
+As a new programmer, inline documentation can be the most helpful. Inline documentation refers to writing comments on the same line as your code. For example if we wrote a line of code to sum 1+1 we might document it as follows:
+
+~~~
+1+1         # adding the numbers 1 and 1 together.
+~~~
+{: .language-python}
+
+Although this is a very simple line of code and it might seem like overkill to document it in this way these types of comments can be very helpful in jogging your memory when returning to a project. Inline comments can also help you to break multi-step programs into digestible and readible pieces.
+
+#### External Documentation
+Sometimes you require more detail than you can comfortably fit in your inline documentation. In this case it can be helpful to create separate files to document your project. This type of documentation will typically focus on the goals, scope, and any special instructions relating to your project rather than the details fo your code. The most common type of external documentation is a README file. It is best practice to create a basic README file for any project. A basic README should include:
+- a brief description of the project,
+- any special instructions for installation or use,
+- the authors and any references.
+
```

Comment:
```suggestion

README files are just text files and it is best practice is to save your README file as a `README.md` markdown document. This file format is automatically recognised by code repositories like GitHub, so your README contents are displayed alongside your code repository. 
```

Just in case readers need extra context about how to generate a README.